examples/demos/create_infrastructure_with_storage.pp

```diff
@@ -0,0 +1,321 @@
+  data    => {
+    name                 => 'Test Puppet LE',
+    enclosureUris        => ['/rest/enclosures/0000000000A66101', '/rest/enclosures/0000000000A66102', '/rest/enclosures/0000000000A66103'],
+    enclosureGroupUri    => Oneview_enclosure_group['Test Puppet Enclosure Group']['data']['name'],
+    firmwareBaselineUri  => 'null',
+    forceInstallFirmware => 'false',
+  }
+}
+
+# Power off the Server Hardware after the Server Profile has been applied
+oneview_server_hardware{'Test Server Hardware Power Off':
+    ensure => 'set_power_state',
+    data   => {
+      hostname    => '0000A66101, bay 5',
+      power_state => 'off',
+    },
+}
+
+# Creates ServerProfileTemplate with Storage, Hardware and boot settings
+oneview_server_profile_template{'Test Puppet SPT':
+  ensure  => 'present',
+  require => Oneview_enclosure_group['Test Puppet Enclosure Group'],
+  data    => {
+    name                  => 'Test Puppet SPT',
+    enclosureGroupUri     => Oneview_enclosure_group['Test Puppet Enclosure Group']['data']['name'],
+    serverHardwareTypeUri => 'SY 480 Gen9 1',
+    connectionSettings    =>
+    {
+      manageConnections => true,
+      connections       =>
+      [
+      {
+          id            => 1,
+          networkUri    => Oneview_ethernet_network['Test Puppet Network']['data']['name'],
+          functionType  => 'Ethernet',
+          portId        => 'Mezz 3:1-a',
+          requestedMbps => '2000'
+      },
+      {
+          id            => 2,
+          networkUri    => Oneview_ethernet_network['Test Puppet Network Create1']['data']['name'],
+          functionType  => 'Ethernet',
+          portId        => 'Mezz 3:1-b',
+          requestedMbps => '2000'
+      }
+      ]
+    },
+    boot                  =>
+    {
+      manageBoot => true,
+      order      =>
+      [
+      'CD',
+      'USB',
+      'HardDisk',
+      'PXE'
+      ]
+    },
+    bootMode              =>
+    {
+      manageMode => true,
+      mode       => 'BIOS',
+      secureBoot => 'Disabled'
+    },
+    bios                  =>
+    {
+      manageBios         => true
+    },
+    sanStorage            =>
+    {
+      manageSanStorage  => true,
+      hostOSType        => 'VMware (ESXi)',
+      volumeAttachments =>
+      [
+        {
+          id           => 1,
+          lun          => '',
+          lunType      => 'Auto',
+          storagePaths =>
+          [
+            {
+              connectionId   => 1,
+              isEnabled      => true,
+              targetSelector => 'Auto'
+            },
+            {
+              connectionId   => 2,
+              isEnabled      => true,
+              targetSelector => 'Auto'
+            }
+          ],
+          volume       => {
+            properties => {
+                provisioningType => 'Thin',
+                size             => 1073741824,
+                name             => 'Test Puppet Storage Volume',
```

Comment:
"Test Puppet Storage Volume" creation is missing here since you are creating everything which is being used in this example.